From bcdc930b761a2433624af946ccbddea8d7171807 Mon Sep 17 00:00:00 2001
From: Peter Wu <lekensteyn@gmail.com>
Date: Sat, 28 Sep 2013 12:27:32 +0200
Subject: [PATCH 14/24] hidpp20: update feature list from SetPoint 6.61.15

cases are now sorted by feature ID, some names have been adjusted for
consistency with the documented names.

New feature:
- 0x0002 FeatureInfo
- 0x6120 Touchmouse6120

Renamed fetures:
- IRoot -> Root
- IFeatureSet -> FeatureSet
- IFirmwareInfo -> DeviceFwVersion
- GetDeviceNameType -> DeviceName
- DFUControl -> Dfucontrol
- batteryLevelStatus -> BatteryStatus
- SpecialKeysMSEButtons -> ReprogControls
---
 hidpp20.c | 27 ++++++++++++++-------------
 1 file changed, 14 insertions(+), 13 deletions(-)

--- a/hidpp20.c
+++ b/hidpp20.c
@@ -34,33 +34,34 @@ static
 const char *
 get_feature_name(uint16_t featureId) {
 	switch (featureId) {
-	case 0x0000: return "IRoot";
-	case FID_IFEATURESET: return "IFeatureSet";
-	case 0x0003: return "IFirmwareInfo";
-	case 0x0005: return "GetDeviceNameType";
-	case 0x1000: return "batteryLevelStatus";
-	case 0x1B00: return "SpecialKeysMSEButtons";
-	case 0x1D4B: return "WirelessDeviceStatus";
-	/* from FeaturesSupported.xml */
+	case 0x0000: return "Root";
+	case 0x0001: return "FeatureSet";
+	case 0x0002: return "FeatureInfo";
+	case 0x0003: return "DeviceFwVersion";
+	case 0x0005: return "DeviceName";
+	case 0x0006: return "DeviceGroups";
+	case 0x00C0: return "Dfucontrol";
+	case 0x1000: return "BatteryStatus";
+	case 0x1981: return "Backlight";
+	case 0x1B00: return "ReprogControls";
 	case 0x1B01: return "ReprogControlsV2";
 	case 0x1B03: return "ReprogControlsV3";
+	case 0x1D4B: return "WirelessDeviceStatus";
+	case 0x2001: return "LeftRightSwap";
 	case 0x2100: return "VerticalScrolling";
 	case 0x2120: return "HiResScrolling";
 	case 0x2200: return "MousePointer";
 	case 0x40A0: return "FnInversion";
+	case 0x40A2: return "NewFnInversion";
 	case 0x4100: return "Encryption";
 	case 0x4301: return "SolarDashboard";
 	case 0x4520: return "KeyboardLayout";
-	case 0x00C0: return "DFUControl";
 	case 0x6010: return "TouchpadFwItems";
 	case 0x6011: return "TouchpadSwItems";
-	case 0x2001: return "LeftRightSwap";
-	case 0x1981: return "Backlight";
 	case 0x6012: return "TouchpadWin8FwItems";
-	case 0x40A2: return "NewFnInversion";
 	case 0x6100: return "TouchpadRawXy";
 	case 0x6110: return "TouchmouseRawPoints";
-	case 0x0006: return "DeviceGroups";
+	case 0x6120: return "Touchmouse6120";
 	default: return "unknown";
 	}
 }
